src/client/Microsoft.Identity.Client/Cache/CacheRefreshReason.cs

```diff
@@ -27,6 +27,14 @@ public enum CacheRefreshReason
         /// <summary>
         /// When the token request goes to the identity provider because refresh_in was used and the existing token needs to be refreshed
         /// </summary>
-        ProactivelyRefreshed = 4
+        ProactivelyRefreshed = 4,
+        /// <summary>
+        /// When the token request goes to the identity provider because WithClaims() was used.
+        /// </summary>
+        WithClaims = 5,
+        /// <summary>
+        /// When the token request goes to the identity provider because WithAccessTokenSha256ToRefresh() is used AND the hash matches.
+        /// </summary>
+        TokenRejected = 6
```

Comment:
What does it mean that we are rejecting the token? Who is rejecting it? The comment doesn't seem to fully explain it.

Reply:
I agree with @trwalke, the naming is misleading. Maybe use `ForceRefreshSha256Token`

Reply:
`ForceRefreshSha256Token` seems too specific because the "sha256" part is really just an implementation detail; it is likely that in the future we may have another parameter `WithAccessTokenToRefresh(string entireToken)`. We shouldn't need to invent yet another enum at that time.

On the other hand, given that we seem comfortable using longer than TwoWords convention, we can use longer phrase to answer the "rejected by who". For example, `TokenRejectedByCaller` (which is actually quite precise).

Just my $0.02

Reply:
@rayluo can you please confirm the final name to be used?

Reply:
Added as 

```cs
/// <summary>
/// Indicates that the resource (e.g., Microsoft Graph) has rejected the provided token. 
/// </summary>
TokenRejectedByResource = 7
```